main.go

```diff
@@ -4,39 +4,207 @@ import (
+			Usage:     "Event status to return on match in generated event.",
+			Value:     &plugin.MatchStatus,
+		},
+		&sensu.PluginConfigOption{
+			Path:      "max-bytes",
+			Env:       "CHECK_LOG_MAX_BYTES",
+			Argument:  "max-bytes",
+			Shorthand: "b",
+			Default:   int64(0),
+			Usage:     "Max number of bytes to read (0 means unlimited).",
+			Value:     &plugin.MaxBytes,
+		},
+		&sensu.PluginConfigOption{
+			Path:      "events-api-url",
+			Env:       "CHECK_LOG_EVENTS_API_URL",
+			Argument:  "events-api-url",
+			Shorthand: "u",
+			Default:   "http://localhost:3031/events",
+			Usage:     "Agent Events API URL.",
+			Value:     &plugin.EventsAPI,
+		},
+		&sensu.PluginConfigOption{
+			Path:      "ignore-initial-run",
+			Env:       "CHECK_LOG_IGNORE_INITIAL_RUN",
+			Argument:  "ignore-initial-run",
+			Shorthand: "I",
+			Default:   false,
+			Usage:     "Suppresses alerts for any matches found on the first run of the plugin.",
+			Value:     &plugin.IgnoreInitialRun,
+		},
+		&sensu.PluginConfigOption{
+			Path:      "check-name-tamplate",
+			Env:       "CHECK_LOG_CHECK_NAME_TEMPLATE",
+			Argument:  "check-name-template",
+			Shorthand: "t",
+			Default:   defaultNameTemplate,
+			Usage:     "Check name to use in generated events",
+			Value:     &plugin.CheckNameTemplate,
+		},
+		&sensu.PluginConfigOption{
+			Path:      "disable-event-generation",
+			Env:       "CHECK_LOG_CHECK_DISABLE_EVENT_GENERATION",
+			Argument:  "disable-event-generation",
+			Shorthand: "D",
+			Default:   false,
+			Usage:     "Disable event generation, send results to stdout instead.",
+			Value:     &plugin.DisableEvent,
+		},
+		&sensu.PluginConfigOption{
+			Path:      "reset-state",
+			Env:       "CHECK_LOG_RESET_STATE",
+			Argument:  "reset-state",
+			Shorthand: "r",
+			Default:   false,
+			Usage:     "Allow automatic state reset if match expression changes, instead of failing.",
+			Value:     &plugin.EnableStateReset,
+		},
+		&sensu.PluginConfigOption{
+			Path:      "inverse-match",
+			Env:       "CHECK_LOG_INVERSE_MATCH",
+			Argument:  "inverse-match",
+			Shorthand: "i",
+			Default:   false,
+			Usage:     "Inverse match, only generate alert event if no lines match.",
+			Value:     &plugin.InverseMatch,
+		},
+		&sensu.PluginConfigOption{
+			Path:      "verbose",
+			Argument:  "verbose",
+			Shorthand: "v",
+			Default:   false,
+			Usage:     "Verbose output, useful for testing.",
+			Value:     &plugin.Verbose,
+		},
+		&sensu.PluginConfigOption{
+			Path:      "dry-run",
+			Argument:  "dry-run",
+			Shorthand: "n",
+			Default:   false,
+			Usage:     "Suppress generation of events and report intended actions instead. (implies verbose)",
+			Value:     &plugin.DryRun,
+		},
+	}
 )
 
 // State represents the state file offset
 type State struct {
-	Offset json.Number `json:"offset"`
+	Offset       json.Number `json:"offset"`
+	LastTime     int64       `json:"last_time"`
+	ModTime      int64       `json:"mod_time"`
```

Comment:
It's a bit risky to use large integers in JSON because other tools will interpret them as floats, like jq. It would be better to store RFC-3339/ISO-8601 formatted datetimes anyways, because then we can capture things like TZ.

Reply:
I consider the state file to be an internal implementation detail. it just need to be monotonic like a unix epoch and does not need to be timezone aware or even human readable. It just need to be monotonically increasing.  
In fact I might as well just change the state file to a binary file and not use json at all, probably faster just to read a binary file.

I dislike using the iso standard for strictly computational time keeping because human parsable timestamps continually require external data updating for correct interpretation for corner-cases like positive and negative leap seconds.  There's no need for that.

Reply:
Binary file with seconds since epoch sounds reasonable. Perhaps use `encoding/gob` for convenience.

Reply:
resolved in additional commit.

Reply:
After adding more testing, removed this time fields from the state.
There's a corner case with the state file that we can't seem to handle. 

If the file is rotated and filled again quickly we might not be able to detect that its a new file and the offset needs to be set back to 0.

Reply:
removed from state file in latest commits. 
Logic now just checks to see if file got smaller since last run. If so it resets file read offset to 0.